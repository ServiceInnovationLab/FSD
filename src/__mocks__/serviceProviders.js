import serviceProvider from './serviceProvider';

const serviceProviders = [ serviceProvider, serviceProvider, serviceProvider ];

export default serviceProviders;
