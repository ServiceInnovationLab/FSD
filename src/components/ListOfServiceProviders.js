import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ListOfServiceProviders extends Component {

  render() {
    const { serviceProviders } = this.props;

    return (
      <div>
        {serviceProviders.map((provider, key) => (
          <h2
            key={key}
            // onClick={() => setCategory(category.name)}
            // className={selectedCategory === category.name ? 'selected' : ''}
          >
            A service provider
          </h2>
        ))}
      </div>
    );
  }

}

ListOfServiceProviders.propTypes = {
  serviceProviders: PropTypes.object
};
