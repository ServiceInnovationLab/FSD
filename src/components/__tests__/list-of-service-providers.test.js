import React from 'react';
import { shallow } from 'enzyme';

import ListOfServiceProviders from '../list-of-service-providers';

import serviceProviders from '../../__mocks__/serviceProviders';

describe('header.js', () => {
  it('renders without error', () => {
    shallow(<ListOfServiceProviders serviceProviders={serviceProviders} />);
  });

  describe('props', () => {
    describe('serviceProviders', () => {
      it('displays the providers correctly', () => {
        const component = shallow(
          <ListOfServiceProviders serviceProviders={serviceProviders} />
        );

        expect(component.find('h2')).toHaveLength(serviceProviders.length);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
