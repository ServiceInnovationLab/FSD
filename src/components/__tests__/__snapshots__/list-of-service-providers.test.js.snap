// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header.js props serviceProviders displays the providers correctly 1`] = `
<div>
  <h2
    key="provider_0"
  >
    A service provider
  </h2>
  <h2
    key="provider_1"
  >
    A service provider
  </h2>
  <h2
    key="provider_2"
  >
    A service provider
  </h2>
</div>
`;
